recipe/meta.yaml

```diff
@@ -31,10 +31,12 @@ requirements:
     - sed               # [unix]
     - openjdk >=8,<17   # [unix]
     - libabseil         # [unix]
-    - grpc-cpp          # [unix]
-    # Remove this pin again once the java parts of bazel can build with a newer gRPC
-    - grpc_java_plugin 1.37  # [osx or (build_platform != target_platform)]
```

Comment:
TBH, I have no idea for which range this is the case. We only have libprotobuf 3.21 builds for grpc>=1.46, so I'm not sure there's even some overlap with the range supported by bazel 4.2.2

Perhaps @xhochy can help?

Reply:
`grpc_java_plugin` doesn't have a dependency on `grpc-cpp`. We can keep this pin for as long as the Java code of bazel is build with this old `grpc-java` version. The relevant parts are forward-compatible and not actually used by any Bazel consumer (I know of) in conda-forge.

Reply:
Thanks. That means this would presumably need https://github.com/conda-forge/grpc_java_plugin-feedstock/pull/45?

I'm not that interested in bazel 4.2 anymore since the feedstocks I'm involved with have moved on

Reply:
Unfortunately, old grpc_java_plugin with new protobuf does not work:
```
compiler/src/java_plugin/cpp/java_generator.cpp:25:10: fatal error: google/protobuf/compiler/java/java_names.h: No such file or directory
   25 | #include <google/protobuf/compiler/java/java_names.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
Target //compiler:grpc_java_plugin failed to build
```

Reply:
> `grpc_java_plugin` doesn't have a dependency on `grpc-cpp`

But it does have one on protobuf... So I'm a bit stuck on how to unblock this.